FirebaseDatabase/Sources/Api/FIRDatabaseQuery.m

```diff
@@ -581,6 +581,13 @@ - (void)keepSynced:(BOOL)keepSynced {
     });
 }
 
+- (void)getDataWithCompletionBlock:(void (^)(NSError *__nullable error,
+                                             FIRDataSnapshot *snapshot))block {
+    dispatch_async([FIRDatabaseQuery sharedQueue], ^{
+      [self.repo getValue:self withCompletionBlock:block];
+    });
```

Comment:
Unrelated - but it looks like Android is inconsistent here. We should probably update Android to also do the dispatch (the `scheduleNow` call) in `Query.java` instead of in the Repo class.